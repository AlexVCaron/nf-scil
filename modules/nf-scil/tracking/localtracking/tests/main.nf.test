nextflow_process {

    name "Test Process TRACKING_LOCALTRACKING"
    script "../main.nf"
    process "TRACKING_LOCALTRACKING"

    tag "modules"
    tag "modules_nfcore"
    tag "tracking"
    tag "tracking/localtracking"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "tracking.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("stracking - localtracking - fa") {
        config "./nextflow_fa.config"
        when {
            process {
                """
             input[0] = LOAD_DATA.out.test_data_directory
                     .map{ test_data_directory -> [
                     [ id:'test', single_end:false ], // meta map
                     file("\${test_data_directory}/map_wm.nii.gz"),
                     file("\${test_data_directory}/fodf.nii.gz"),
                     file("\${test_data_directory}/fa.nii.gz")
             ]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stracking - localtracking - wm") {
        config "./nextflow_wm.config"
        when {
            process {
                """
             input[0] = LOAD_DATA.out.test_data_directory
                     .map{ test_data_directory -> [
                     [ id:'test', single_end:false ], // meta map
                     file("\${test_data_directory}/map_wm.nii.gz"),
                     file("\${test_data_directory}/fodf.nii.gz"),
                     file("\${test_data_directory}/fa.nii.gz")
             ]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
