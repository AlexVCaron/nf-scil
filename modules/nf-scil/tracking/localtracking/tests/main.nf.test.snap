{
    "tracking - localtracking - wm": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking.trk:md5,c5382df437a30af947d142345141de32"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking_config.json:md5,5eac37e6262a85d8d1b33c92e1f35796"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_seeding_mask.nii.gz:md5,6180aeda681e4443e56287ef1e1aee83"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking_mask.nii.gz:md5,6180aeda681e4443e56287ef1e1aee83"
                    ]
                ],
                "4": [
                    "versions.yml:md5,821283cb3ddc035c572b928a078cd98a"
                ],
                "config": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking_config.json:md5,5eac37e6262a85d8d1b33c92e1f35796"
                    ]
                ],
                "seedmask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_seeding_mask.nii.gz:md5,6180aeda681e4443e56287ef1e1aee83"
                    ]
                ],
                "trackmask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking_mask.nii.gz:md5,6180aeda681e4443e56287ef1e1aee83"
                    ]
                ],
                "trk": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking.trk:md5,c5382df437a30af947d142345141de32"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,821283cb3ddc035c572b928a078cd98a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T19:10:10.242957"
    },
    "tracking - localtracking - fa": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking.trk:md5,bc4fb50d8e3c21902f11bff986fd84f4"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking_config.json:md5,a082f8d2f2b92cb3f2912a796942a786"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_seeding_mask.nii.gz:md5,0484a2e44ff153caa79819a8646cc33c"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking_mask.nii.gz:md5,0484a2e44ff153caa79819a8646cc33c"
                    ]
                ],
                "4": [
                    "versions.yml:md5,821283cb3ddc035c572b928a078cd98a"
                ],
                "config": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking_config.json:md5,a082f8d2f2b92cb3f2912a796942a786"
                    ]
                ],
                "seedmask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_seeding_mask.nii.gz:md5,0484a2e44ff153caa79819a8646cc33c"
                    ]
                ],
                "trackmask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking_mask.nii.gz:md5,0484a2e44ff153caa79819a8646cc33c"
                    ]
                ],
                "trk": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__local_tracking.trk:md5,bc4fb50d8e3c21902f11bff986fd84f4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,821283cb3ddc035c572b928a078cd98a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T19:01:42.53971"
    }
}